packages/flutter/lib/src/animation/curves.dart

```diff
@@ -468,6 +469,45 @@ class ElasticInOutCurve extends Curve {
   }
 }
 
+/// A curve that progresses linearly until a specified [startingPoint], at which
+/// point [curve] will begin. Unlike [Interval], [curve] will not start at zero,
+/// but will use [startingPoint] as the Y position.
+class SuspendedCurve extends Curve {
```

Comment:
Expanded in the class docs with the new commits.

I wasn't able to keep SuspendedCurve private in bottom_sheet.dart, since it is also used in scaffold.dart for persistent bottom sheets, so I moved it to bottom_sheet.dart and renamed it as BottomSheetSuspendedCurve.